### PR TITLE
add github action

### PR DESCRIPTION
**Pull request by yunnysunny**
Since the trvais ci is not free now,  and the test on travis is all been skiped. I rewrite the CI use github action.

It 's import to merge this request , since the tests is not tirggered when new code was merge to master.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        os: [ubuntu-latest]
+        include:
+          - node: 14
+            os: macos-latest
+            CPPFLAGS: "-I/usr/local/opt/openssl/include"
+            LDFLAGS: "-L/usr/local/opt/openssl/lib"
+          - node: 16
+            os: windows-2019
+    steps:
+      - uses: actions/checkout@v2
```
Comment (cclauss) [inline .github/workflows/ci.yml:23]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        with:
+          submodules: recursive
+
+      - name: Install Node - ${{ matrix.node }} in ${{ runner.os }}
+        uses: actions/setup-node@v2
```
Comment (cclauss) [inline .github/workflows/ci.yml:28]:
```suggestion
        uses: actions/setup-node@v3
```

**Review by cclauss — APPROVED** [review]
These tests did not yet pass at https://github.com/yunnysunny/node-rdkafka/actions

**Comment by cclauss** [comment]
Node.js 8 seems to fail…. https://github.com/yunnysunny/node-rdkafka/actions

**Comment by cclauss** [comment]
Node.js 8 seems to fail…. https://github.com/yunnysunny/node-rdkafka/actions

**Comment by yunnysunny** [comment]
> Node.js 8 seems to fail…. https://github.com/yunnysunny/node-rdkafka/actions

Need to update node-gyp to a correct version on Node 8. I will try the 7.x version.

**Comment by cclauss** [comment]
Awesome!!  Extra credit? Node.js 18 on Windows 22

**Comment by yunnysunny** [comment]
> Awesome!! Extra credit? Node.js 18 on Windows 22

Node-gyp does not work on Windows 2022, https://github.com/yunnysunny/node-rdkafka/runs/7162913449?check_suite_focus=true . When I run update command on Windows 2022, it failed https://github.com/yunnysunny/node-rdkafka/runs/7162957526?check_suite_focus=true

**Comment by GaryWilber** [comment]
This was partly done in #978 and I added Windows & Mac to the matrix in #998. I think this can be closed now?

**Pull request closed without merging**